packages/lib/navigateInTopWindow.ts

```diff
@@ -0,0 +1,5 @@
+export const navigateInTopWindow = (url: string) => {
+  // In some weird unknown case, window.top.location could be null as per typescript. In such case, we still do the redirect but in same window
+  const location = window.top ? window.top.location : window.location;
+  location.href = url;
```

Comment:
As discussed in campsite, the problem is because our iframe with src=app.cal.com is nested in another iframe with src=xyz.com that is used on xyz.com domain page.

That means that our iframe is a cross-origin iframe - xyz.com loads iframe from app.cal.com, which means that `window.top` will be defined, but `window.top.location` is not accessible to cross-origin iframes, so our iframe won't have access to it and trying to access it might result in error.

So we need to update the check:
```
export const navigateInTopWindow = (url: string) => {
  try {
    if (window.top && window.top.location) {
      window.top.location.href = url;
    } else {
      window.location.href = url;
    }
  } catch (error) {
    window.location.href = url;
  }
};
```

Reply:
Cal iframe being cross-origin in the first place was the reason that `window.parent.location.href` was giving permission error:

<img width="351" alt="Screenshot 2024-07-08 at 10 56 23" src="https://github.com/calcom/cal.com/assets/42170848/b4378901-f5f4-4794-9a88-fa47b0cf6d31">

which is why when embed is used outside app.cal.com (which i guess is 100% of the times) we can only use `window.location.href` to only redirect the cal iframe itself, not the whole parent page, which as i understand is not possible.

Reply:
Interesting, appreciate the loom and PR looks good !